src/cmd-oscontainer

```diff
@@ -0,0 +1,139 @@
+                        tls_verify=True):
+    dest = os.path.realpath(dest)
+    subprocess.check_call(["ostree", "--repo="+dest, "refs"])
+    rootarg = '--root='+containers_storage
+    if not tls_verify:
+        tls_arg = '--tls-verify=false'
+    else:
+        tls_arg = '--tls-verify'
+    run_verbose(['podman', rootarg, 'pull', tls_arg, src])
+    inspect = run_get_json(['podman', rootarg, 'inspect', src])[0]
+    commit = inspect['Labels'].get(OSCONTAINER_COMMIT_LABEL)
+    if commit is None:
+        raise SystemExit("Failed to find label '{}'".format(OSCONTAINER_COMMIT_LABEL))
+    iid = inspect['Id']
+    print("Preparing to extract cid {}", iid)
+    # We're not actually going to run the container. The main thing `create` does
+    # then for us is "materialize" the merged rootfs, so we can mount it.
+    # In theory we shouldn't need --entrypoint=/enoent here, but
+    # it works around a podman bug.
+    cid = run_get_string(['podman', rootarg, 'create', '--entrypoint=/enoent', iid])
+    mnt = run_get_string(['podman', rootarg, 'mount', cid])
+    try:
+        src_repo = mnt+'/srv/repo'
+        run_verbose(["ostree", "--repo="+dest, "pull-local", src_repo, commit])
+    finally:
+        subprocess.call(['podman', rootarg, 'umount', cid])
+
+# Given an OSTree repository at src (and exactly one ref) generate an oscontainer
+# with it.
+def oscontainer_build(containers_storage, src, ref, image_name_and_tag,
+                      base_image='scratch',
+                      push=False, tls_verify=True):
+    r = OSTree.Repo.new(Gio.File.new_for_path(src))
+    r.open(None)
+
+    [_, rev] = r.resolve_rev(ref, True)
+    if ref != rev:
+        print("Resolved {} = {}".format(ref, rev))
+    [_, ostree_commit, _] = r.load_commit(rev)
+    ostree_commitmeta = ostree_commit.get_child_value(0)
+    versionv = ostree_commitmeta.lookup_value("version", GLib.VariantType.new("s"))
+    if versionv:
+        ostree_version = versionv.get_string()
+    else:
+        ostree_version = None
+
+    rootarg = '--root='+containers_storage
+    bid = run_get_string(['buildah', rootarg, 'from', base_image])
+    mnt = run_get_string(['buildah', rootarg, 'mount', bid])
+    try:
+        dest_repo = mnt+'/srv/repo'
+        subprocess.check_call(['mkdir', '-p', dest_repo])
+        subprocess.check_call(["ostree", "--repo="+dest_repo, "init", "--mode=archive"])
+        # Note that oscontainers don't have refs
+        print("Copying ostree commit into container: {} ...".format(rev))
+        run_verbose(["ostree", "--repo="+dest_repo, "pull-local", src, rev])
+
+        # We use /noentry to trick `podman create` into not erroring out
+        # on a container with no cmd/entrypoint.  It won't actually be run.
+        config=['--entrypoint', '["/noentry"]',
+                '-l', OSCONTAINER_COMMIT_LABEL+'='+rev]
+        if ostree_version is not None:
+            config += ['-l', 'version='+ostree_version]
```

Comment:
Why not e.g. `com.coreos.ostree-version` here?

Reply:
There's a standard container `version` field